ddl/partition.go

```diff
@@ -1206,3 +1208,25 @@ func truncateTableByReassignPartitionIDs(t *meta.Meta, tblInfo *model.TableInfo)
 	tblInfo.Partition.Definitions = newDefs
 	return nil
 }
+
+func onAlterTablePartition(t *meta.Meta, job *model.Job) (int64, error) {
+	var rules []*placement.Rule
+	err := job.DecodeArgs(&rules)
+	if err != nil {
+		job.State = model.JobStateCancelled
+		return 0, errors.Trace(err)
+	}
+
+	ver, err := t.GetSchemaVersion()
+	if err != nil {
+		return ver, errors.Trace(err)
+	}
+
+	if err := infosync.UpdatePlacementRules(nil, rules); err != nil {
```

Comment:
It seems we needn't sync this information to other TiDB. Could we process`alter table tblName alter partition partiName add placement...` as an ordinary executor instead of DDL?

Reply:
@djshow832 Need help for this question.

According to RFC, we need to drop rules when drop/truncate partition. So we will have `infosync` in DDL anyway. And mentor mentioned that there should be some state for `SHOW PLACEMENT` statements. IMO, better stick to DDL.